sdk/python/feast/field.py

```diff
@@ -25,31 +27,43 @@ class Field:
     Attributes:
         name: The name of the field.
         dtype: The type of the field, such as string or float.
+        tags: User-defined metadata in dictionary form.
     """
 
     name: str
     dtype: FeastType
+    tags: Dict[str, str]
 
     def __init__(
-        self, *, name: str, dtype: FeastType,
+        self,
+        *,
+        name: Optional[str] = None,
+        dtype: Optional[FeastType] = None,
+        tags: Optional[Dict[str, str]] = None,
     ):
         """
         Creates a Field object.
 
         Args:
             name: The name of the field.
             dtype: The type of the field, such as string or float.
+            tags (optional): User-defined metadata in dictionary form.
         """
-        self.name = name
-        self.dtype = dtype
+        self.name = name or ""
+        self.dtype = dtype or PrimitiveFeastType.INVALID
+        self.tags = tags or {}
 
     def __eq__(self, other):
-        if self.name != other.name or self.dtype != other.dtype:
+        if (
+            self.name != other.name
+            or self.dtype != other.dtype
+            or self.tags != other.tags
+        ):
             return False
         return True
 
     def __hash__(self):
-        return hash((self.name, hash(self.dtype)))
+        return hash((self.name, hash(self.dtype), hash(self.tags)))
```

Comment:
I don't think you can hash a `Dict`? it's fine to just leave this out

Reply:
Removed.